internal/datacoord/compaction_trigger_v2.go

```diff
@@ -91,9 +93,11 @@ func (m *CompactionTriggerManager) BuildLevelZeroCompactionPlan(view CompactionV
 	var segmentBinlogs []*datapb.CompactionSegmentBinlogs
 	levelZeroSegs := lo.Map(view.GetSegmentsView(), func(v *SegmentView, _ int) *datapb.CompactionSegmentBinlogs {
 		s := m.meta.GetSegment(v.ID)
+		cloned := s.Clone()
+		binlog.DecompressBinLog(storage.DeleteBinlog, cloned.GetCollectionID(), cloned.GetPartitionID(), cloned.GetID(), cloned.GetDeltalogs())
```

Comment:
DecompressBinLog in dn(compact func in internal/datanode/l0_compactor.go) may be more suitable, since dc not use path anymore.